stereo_points/stereo_points.py

```diff
@@ -107,205 +104,194 @@ def setup(self):
+            self.updatePointsCoordsFromXYZ(self.GetCoordTable(), self.referenceImage_selectionCombo.currentNode(), newNode)
 
     def onControlPointSelectedChanged(self, newNode):
-        # print('selection changed !')
-        if (type(newNode) == type(slicer.vtkMRMLMarkupsLineNode())):
+        # for each markupLineNode, we maintain a table with the coord conversion.
+        # table nodes are names with the name of the markupLine with _coordsConversion as prefix
+        if type(newNode) == type(slicer.vtkMRMLMarkupsLineNode()):
             coordTableName = newNode.GetName() + "_coordsConversion"
         else:
             return
-        # print('looking for '+coordTableName)
-
-        if slicer.mrmlScene.GetNodesByName(coordTableName).GetNumberOfItems() == 0:
-            # print('create new table')
-            coordTable = slicer.vtkMRMLTableNode()
-            coordTable.SetName(coordTableName)
-            coordTable.SetLocked(True)
-            for col in ['Marker', 'x', 'y', 'z', 'r', 'a', 'd', 'X', 'Y', 'Z', 'R', 'A', 'S', 'i', 'j', 'k']:
-                c = coordTable.AddColumn()
+
+        coordTable_id = newNode.GetNodeReferenceID("stereotaxia_coordTable")
+
+        # if the lineMarkup does not reference any coordtable yet, create it
+        if coordTable_id is None:
+            # prepare a new table
+            newCoordTable = slicer.vtkMRMLTableNode()
+            newCoordTable.SetName(coordTableName)
+            newCoordTable.SetLocked(True)
+            # columns:
+            # x, y, z, r, a, d: leksell frame settings + depth
+            # XYZ cartesian coordinates in leksell space
+            # RAS coordinates in physical space (what slicer uses to place the points)
+            # ijk coordinates in image space
+            for col in ["Marker", "x", "y", "z", "r", "a", "d", "X", "Y", "Z", "R", "A", "S", "i", "j", "k"]:
+                c = newCoordTable.AddColumn()
                 c.SetName(col)
 
-            slicer.mrmlScene.AddNode(coordTable)
-            # keep a map of the fiducialNode <-> TableNode couples
-            self.fiducialNodesToCoordTables_id_tuples.append((newNode.GetID(), coordTable.GetID()))
+            slicer.mrmlScene.AddNode(newCoordTable)
+            # refererence the line from the table so we can get back to it
+            newCoordTable.AddNodeReferenceID("stereotaxia_trajLine", newNode.GetID())
+            newNode.AddNodeReferenceID("stereotaxia_coordTable", newCoordTable.GetID())
             # add an observer for both nodes, whenever one is renamed, the other one is renamed as well
-            coordTable.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
+            self.lObs_tableNode = newCoordTable.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
             newNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onControlPointNodeModified)
```

Comment:
Same applies here as to the table node observation. You have to remove the old observer when the user selects a different fiducial node (and at module cleanup).